### ISSUE TITLE
bug(material/datepicker): Voiceover Loses Focus when pressing Page Down key

### ISSUE THREAD

**Issue by zarend** [angular/components#24330]
When navigating the calendar with VoiceOver enabled, pressing Page Down causes the screenreader focus to move to the entire calendar, instead of stay on an individual date.

#### Reproduction

Steps to reproduce:
1. With Screenreader enabled, Open the [datepicker expands in a browser](https://material.angular.io/components/datepicker/examples) and navigate to the Basic Datepicker Demo.
2. Click on the Calendar button to open the datepicker 
3.  navigate to January 28, 2022
4. (VoiceOver reads "January 28, 2022". the cell for Jan 28, 2022 has blue focus outline and black voiceover outline
5. Press PageDown
6. (VoiceOver reads "February, 2022". The Cell for Feb 28, 2022 has blue focus outline. Voiceover's black outline is around the entire box. Voiceover should read "February 28, 2022".

 
#### Expected Behavior
Voiceover reads "February 28, 2022".

#### Actual Behavior
voiceover reads "February, 2022"

#### Environment
Operating System | macOS Monterey 12.1 (21C52)
Browser | Chrome Version 97.0.4692.99 (Official Build) (x86_64), Safari Version 15.3 (17612.4.9.1.5)
Screenreader | VoiceOver

https://user-images.githubusercontent.com/7720245/152051557-abb11c7d-3b66-43b7-bd1a-41c517cab896.mov

##### Additional Notes
This is copied from an [internal bug report](http://b/216644357) found during a11y testing.

**Comment by zarend** [comment]
@Splaktar I'm stumped on this one. I'm curious if you've come across anything similar to this? This seems vaguely familiar to problems we've had with setting focus and screenreader announcements when opening dialogs?

My current theory, is that VoiceOver gets confused when the focused element is removed from the DOM, which causes it to focus on the entire table.

**Comment by zarend** [comment]
After talking to @crisbeto , looks like the aria-live region is interfering with announcing the calendar body. I'll keep looking into this...

**Comment by Splaktar** [comment]
It sounds like you are on the right track. I don't think that I recall any problems (or solutions) to something similar.
